webapp/src/shared/components/Text/Text.tsx

```diff
@@ -0,0 +1,52 @@
+import './Text.css';
+
+export class TextVariant {
+  public static TITLE = new TextVariant('h1', '3rem');
+  public static SUBTITLE = new TextVariant('h2', '2.5rem');
+  public static HEADING = new TextVariant('h3', '1.5rem');
+  public static SUBHEADING = new TextVariant('h4', '1.25rem');
+  public static PARAGRAPH = new TextVariant('p', '1rem');
+  public static CAPTION = new TextVariant('p', '0.875rem');
+
+  private constructor(public tag: string, public size: string) {}
+}
+
+export enum TextColor {
+  WARNING = 'text-warning',
+  SUBMIT = 'text-submit',
+  ONLINE = 'text-online',
+  OFFLINE = 'text-offline',
+  LIGHT = 'text-light',
+  BACKGROUND = 'text-background',
+  DARK = 'text-dark',
+  GAME = 'text-game',
+}
```

Comment:
What do you think about using this for all components and moving it to `webapp/src/shared/types.ts`?

We could get rid of shared `enum Color` and move classes to root index.css. We could just use classes for colors. This would avoid having duplicated denominations.

If we have problems with special fills such as the gradient, we can specify in css custom chainings. Lets say we cannot use the same properties for an icon and a header. We could do something like:

```css
svg.text-game {/* custom properties */}
h1.text-game {/* custom properties */}
```

Reply:
In my opinion, using classes for the colors makes sense, and I've updated the `Colors` enum.

I prefer not to add all the colors from this component to `index.css` because the gradient is specific to the text.

Also, I don't think there is an easy way to add a gradient to an SVG without creating additional SVGs.

I've created an "extended" enum for the `TextColor`, which combines the `Color` with a `CustomTextColor` for the gradient.

It is a bit of a hack, but it works :sweat_smile: - https://github.com/microsoft/TypeScript/issues/17592#issuecomment-449440944

What do we think about this?